src/mobu/services/flock.py

```diff
@@ -209,11 +217,22 @@ def _create_monkey(self, user: AuthenticatedUser) -> Monkey:
     async def _create_users(self) -> list[AuthenticatedUser]:
         """Create the authenticated users the monkeys will run as."""
         users = self._config.users
+        if users and self._replica_count > 1:
+            raise RuntimeError(
+                "Only user_spec users are allowed if replica_count > 1"
+            )
         if not users:
             if not self._config.user_spec:
                 raise RuntimeError("Neither users nor user_spec set")
```

Comment:
Actually, any objections to moving all of this validation into the config model? We know it all at config time.